extensions/android-paging3/src/test/java/com/squareup/sqldelight/android/paging3/KeyedQueryPagingSourceTest.kt

```diff
@@ -0,0 +1,187 @@
+package com.squareup.sqldelight.android.paging3
+
+import androidx.paging.PagingSource.LoadParams.Refresh
+import androidx.paging.PagingSource.LoadResult
+import com.squareup.sqldelight.Query
+import com.squareup.sqldelight.Transacter
+import com.squareup.sqldelight.TransacterImpl
+import com.squareup.sqldelight.android.AndroidSqliteDriver
+import com.squareup.sqldelight.db.SqlDriver
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+import kotlin.test.assertEquals
+
+@ExperimentalCoroutinesApi
+@RunWith(RobolectricTestRunner::class)
+class KeyedQueryPagingSourceTest {
+
+  private lateinit var driver: SqlDriver
+  private lateinit var transacter: Transacter
+
+  @Before
+  fun before() {
+    driver = AndroidSqliteDriver(
+      object : SqlDriver.Schema {
+        override val version: Int = 1
+
+        override fun create(db: SqlDriver) {
+          db.execute(null, "CREATE TABLE testTable (value INTEGER PRIMARY KEY)", 0)
+
+          for (i in 0L until 10L) {
+            insert(i, db)
+          }
+        }
+
+        override fun migrate(
+          db: SqlDriver,
+          oldVersion: Int,
+          newVersion: Int
+        ) {
+          throw AssertionError("DB Migration shouldn't occur")
+        }
+      },
+      RuntimeEnvironment.application
```

Comment:
```suggestion
      getApplicationContext()
```

I recently changed all these throughout the repo in #2217

Reply:
Awesome, I'll update the next time I make some changes.

Reply:
Updated to pure JVM for both offset and Keyed :)